core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java

```diff
@@ -199,13 +237,25 @@ public void setCredentialsProviders(Collection<? extends AwsCredentialsProvider>
         }
 
         @Override
-        public Builder credentialsProviders(AwsCredentialsProvider... credentialsProviders) {
-            return credentialsProviders(Arrays.asList(credentialsProviders));
+        public Builder credentialsIdentityProviders(
+                Collection<? extends IdentityProvider<? extends AwsCredentialsIdentity>> credentialsProviders) {
```

Comment:
General comment on use of Collection , if user is doing Set.of("1","2", "3") the API will accept it but the order will not be maintained.
I think it should be `List` if order is important,

Reply:
Hmm, I kinda see the point. I don't know the history of why List wasn't used. Maybe it was to give more flexibility to the caller. We wouldn't change the signature now anyways. This change is just adding overloaded usage for new type and we wouldn't want it to use List where the other one allows Collection.